File: assign3/MovieDescription.cpp
#include "MovieDescription.hpp"

MovieDescription::MovieDescription() {}
MovieDescription::MovieDescription(string setTitle, string setRated, string setReleased, string setRuntime,
                                   string setPlot, string setFilename, string setGenre, string setActors) {
    title = setTitle;
    rated = setRated;
    released = setReleased;
    runtime = setRuntime;
    plot = setPlot;
    filename = setFilename;
    genre = setGenre;
    actors = setActors;
}
MovieDescription::~MovieDescription() {}

void MovieDescription::setTitle(string setTitle) {title = setTitle;}
void MovieDescription::setRated(string setRated) {rated = setRated;}
void MovieDescription::setReleased(string setReleased) {released = setReleased;}
void MovieDescription::setRuntime(string setRuntime) {runtime = setRuntime;}
void MovieDescription::setPlot(string setPlot) {plot = setPlot;}
void MovieDescription::setFilename(string setFilename) {filename = setFilename;}
void MovieDescription::setGenre(string setGenre) {genre = setGenre;}
void MovieDescription::setActors(string setActors) {actors = setActors;}

string MovieDescription::getTitle()const {return title;}
string MovieDescription::getRated()const {return rated;}
string MovieDescription::getReleased()const {return released;}
string MovieDescription::getRuntime()const {return runtime;}
string MovieDescription::getPlot()const {return plot;}
string MovieDescription::getFilename()const {return filename;}
string MovieDescription::getGenre()const {return genre;}
string MovieDescription::getActors()const {return actors;}

string MovieDescription::toString() {
    string str;

    str.append(this->getTitle());
    str.append("\n    Rated: "+this->getRated());
    str.append("\n    Released: "+this->getReleased());
    str.append("\n    Runtime: "+this->getRuntime());
    str.append("\n    Plot: "+this->getPlot());
    str.append("\n    Filename: "+this->getFilename());
    str.append("\n    Genre: "+this->getGenre());
    str.append("\n    Actors: "+this->getActors());

    return str;
}

File: SER321-master/Assigns/Assign3/src/java/MovieDescription.java
import java.io.Serializable;
import java.util.List;
import java.util.ArrayList;
import org.json.*;

public class MovieDescription
{
  private String title;
  private String rated;
  private String released;
  private String runtime;
  private String plot;
  private String filename;
  private ArrayList<String> genre;
  private ArrayList<String> actors;

  public MovieDescription(String title, String rated, String released, String runtime, String plot, String filename, ArrayList<String> genre, ArrayList<String> actors)
  {
    setTitle(title);
    setRated(rated);
    setReleased(released);
    setRuntime(runtime);
    setPlot(plot);
    setFilename(filename);
    setGenre(genre);
    setActors(actors);
  }

  public MovieDescription(JSONObject jsonObj)
  {
    released = jsonObj.getString("Released");
    rated = jsonObj.getString("Rated");
    actors = toArrayList(jsonObj.getJSONArray("Actors"));
    plot = jsonObj.getString("Plot");
    runtime = jsonObj.getString("Runtime");
    genre = toArrayList(jsonObj.getJSONArray("Genre"));
    filename = jsonObj.getString("Filename");
    title = jsonObj.getString("Title");
  }

  public JSONObject toJSONObject()
  {
     JSONObject obj = new JSONObject();
     obj.put("Released", released);
     obj.put("Rated", rated);
     obj.put("Actors", actors);
     obj.put("Plot", plot);
     obj.put("Runtime", runtime);
     obj.put("Genre", genre);
     obj.put("Filename", filename);
     obj.put("Title", title);
     //System.out.println("User toJSONObject returning: "+obj.toString());
     return obj;
  }

  public void setTitle(String title)
  {
    this.title = title;
  }

  public void setRated(String rated)
  {
    this.rated = rated;
  }

  public void setReleased(String released)
  {
    this.released = released;
  }
  public void setRuntime(String runtime)
  {
    this.runtime = runtime;
  }

  public void setPlot(String plot)
  {
    this.plot = plot;
  }

  public void setFilename(String filename)
  {
    this.filename = filename;
  }

  public void setGenre(ArrayList<String> genre)
  {
    this.genre = genre;
  }
  public void setActors(ArrayList<String> actors)
  {
    this.actors = actors;
  }

  public void print()
  {
    System.out.println("\tRated " + rated);
    System.out.println("\tReleased " + released);
    System.out.println("\tRuntime " + runtime);
    System.out.println("\tPlot: " + plot);
    System.out.println("\tFilename " + filename);
    System.out.println("\tGenre: " + genre);
    System.out.println("\tActors: " + actors);
  }

  public String getTitle()
  {
    return title;
  }

  public String getRated()
  {
    return rated;
  }

  public String getReleased()
  {
    return released;
  }

  public String getRuntime()
  {
    return runtime;
  }

  public String getPlot()
  {
    return plot;
  }

  public String getFilename()
  {
    return filename;
  }

  public ArrayList<String> getGenre()
  {
    return genre;
  }

  public ArrayList<String> getActors()
  {
    return actors;
  }

  private ArrayList<String> toArrayList(JSONArray jsonArray)
  {
    ArrayList<String> list = new ArrayList<String>();

    if (jsonArray != null)
    {
      int length = jsonArray.length();

      for (int i = 0; i < length; i++)
      {
        list.add(jsonArray.get(i).toString());
      }
    }

    return list;
  }
}
